### ISSUE TITLE
Option for disabling formatting of Class and ID literals

### ISSUE THREAD

**Issue by Shinigami92** [prettier/plugin-pug#36]
Let the user decide if the formatter should format Class and ID literals or not

---

based on #32

**Comment by Shinigami92** [comment]
@lehni Do you also want to work on some of these `help wanted` issues?

**Comment by lehni** [comment]
@Shinigami92 at the moment only things that I need for my own use-case of the plugin. Once these things are taken care of, if I find the time and have an idea for how to do it, why not!

**Comment by SkyaTura** [comment]
I pretend to do it this weekend 🤗

**Comment by SkyaTura** [comment]
@Shinigami92 my original intent was to provide 3 options:
- as-is: disables class and id formatting
- use literals: forces all valid classes and ids to be printed as literals
- use attributes: forces all classes and ids to be printed as attributes

_HOWEVER_, as far as I understood, the plugin would need a huge refactor (or an ugly workaround) to allow a new attribute to be set, specially in cases where there is no attribute at all in the element.

I'm considering to release only the first two options, which is already working fine.

What are your thoughts about it?

**Comment by Shinigami92** [comment]
@SkyaTura Yes I also saw this potential of options :smiley:  
The thing is, if we now release it with just enabling/disabling (boolean flag) possibility, we cannot change it later on without a breaking release

So maybe we should keep this potential feature in mind (outsource it to a new issue) and implement your current PR with a `choice` option

**Comment by SkyaTura** [comment]
As soon as I have some time I'll do it ;)

**Comment by Shinigami92** [comment]
Take your time, it's just a new feature and I don't even know if many want and will use this feature ^^'
Would be cool if I could somehow get statistics from it ^^

**Comment by basaran** [comment]
Related to #165

Some editor extensions are having a tough time providing intellisense auto completions (vetur) when the intellisense is invoked inline on a `pug` file.

This is not an issue with the `plugin-pug`, but it would be extremely helpful to have an option to disable inlining of css class names during development.

Option can be called something like `pugTagCss`

Please see the recording:

![inline_classes](https://user-images.githubusercontent.com/30809170/100496518-bb8a8a80-3122-11eb-9125-d3b43e5d0532.gif)

**Comment by Shinigami92** [comment]
@basaran So I invited you with `write` permissions to our repo
This allowes you to create branches and modify branches except the `master` branch
`master` branch is readonly for you

This is useful because you don't need to create a fork and we can work together on this feature

So now you have two options:
1. Wait on @SkyaTura and ask him if you can help him with his PR?
2. Open a second branch and make a competition 😛 The better branch wins 😅

Keep in mind: always have fun and keep friendly 😁

**Comment by Shinigami92** [comment]
Oh, one thing I forgot: If working together on one branch => **DON'T** use `rebase`/`force-push`

**Comment by basaran** [comment]
Arigatou Gozaimasu :) I will be happy to help in anyway I can

**Comment by SkyaTura** [comment]
I was having some hard time re-learning how to type on my new split-ortholinear keyboard, but now that my WPM and accuracy are acceptable again I can finish my work here.

But I'm still open to ideas of how create an attribute when class/id are unset to implement the `attribute` mode.
